src/plot.jl

```diff
@@ -0,0 +1,448 @@
+            condx = gantt.starts[k] <= pos[1] <= gantt.stops[k]
+            condy = gantt.threads[k] - 0.3 <= pos[2] <= gantt.threads[k] + 0.3
+            if condx && condy
+                match = true
+                tasklog.label != "" && (to[] = "$(tasklog.label)")
+            end
+
+            k += 1
+        end
+
+        !match && (to[] = "Task Label")
+
+    end
+    on(to) do t
+        ax.title = "Parallel Trace\n$t"
+    end
+end
+
+
+"""
+    plot(logger; categories)  
+Plot DataFlowTasks `logger` labeled informations with categories.  
+Note : if there's too many tasks, the DAG won't be plotted,
+although you can manually call `dagplot(logger)`.
+
+## Example  
+Use the above exemple with
+```jldoctest
+using GLMakie, GraphViz
+using DataFlowTasks
+using DataFlowTasks: RW
+import DataFlowTasks as DFT
+
+computing(A) = exp.(sum(A).^2).^2
+function work(A, B)
+    @dspawn computing(@RW(A)) label="A"
+    @dspawn computing(@RW(B)) label="B"
+    @dspawn computing(@RW(A)) label="A"
+    @dspawn computing(@RW(B)) label="B"
+end
+
+# Context
+A = ones(1000, 1000)
+B = ones(1000, 1000)
+
+# Compilation
+work(copy(A), copy(B))
+
+# Reset logger and taskcounter
+DFT.resetlogger!()
+DFT.TASKCOUNTER[] = 0
+
+# Reak Work
+work(A, B)
+
+# Logger Visualization
+DFT.plot(DFT.getlogger(), categories=["A", "B"])
+```
+"""
+function plot(logger::Logger; categories=String[])
+    # Figure
+    # ------
+    fig = Figure(
+        backgroundcolor = RGBf(0.98, 0.98, 0.98),
+        resolution = (1280, 720)
+    )
+
+    # Extract logger informations
+    # ---------------------------
+    loginfo = LoggerInfo(logger, categories, criticalpath())
+    gantt = Gantt()
+    extractloggerinfo!(logger, loginfo, gantt)
+    shouldplotdag = (nbtasknodes(logger) < 75)  # arbitrary limit
+
+    # Layouts (conditionnal depending on DAG size)
+    # --------------------------------------------
+    axtrc = Axis(fig[1,1]     , title="Parallel Trace\n Task Label")
+    axact = Axis(fig[2,1][1,1], title="Activity")
+    axinf = Axis(fig[2,1][1,2], title="Time Bounds")
+    axcat = Axis(fig[2,1][1,3], title="Times per Category")
+    if !shouldplotdag
+        @info "The dag has too many nodes, plot it separately with `dagplot()` so it can be readable"
+    else
+        axdag = Axis(fig[1:2,2]   , title="Graph")
+        colsize!(fig.layout, 1, Relative(3/4))
+    end
+    # -------
+    rowsize!(fig.layout, 1, Relative(2/3))
+
+    # Plot each part
+    # --------------
+    traceplot(axtrc, logger, gantt, loginfo)
+    activityplot(axact, loginfo)
+    boundsplot(axinf, loginfo)
+    categoriesplot(axcat, loginfo)
+    shouldplotdag && dagplot(axdag, logger)
+
+    # Events management
+    react(axtrc, logger, gantt)
+
+    # Terminal Informations
+    # ---------------------
+    @info "Computing    : $(loginfo.computingtime)"
+    @info "Inserting    : $(loginfo.insertingtime)"
+    @info "Other        : $(loginfo.othertime)"
+
+    fig
```

Comment:
It's a good idea to return `fig`. It means that in automated tests, we could do something like:
```
fig = plot(getlogger(), categories=["A", "B"])
save("logger.png", fig)
```

At least when this runs on a local machine, we can check the results. And the generated figure could be included in the documentation.

Reply:
In relation to the doctest conversation above, I'm wondering whether it would make sense to amend the test in `logger_test.jl` to call `plot` and `savefig`.
During CI, this would at least ensure that the code runs. When running the tests locally, we could then look at the generated figure to check that it looks good. Would that make sense?